### PR TITLE
Invalidate page cache with Xen after a page is written to

### PR DESCRIPTION
**Pull request by tklengyel**

### CODE REVIEW

File: libvmi/driver/memory_cache.c [hunk 1]
```diff
@@ -297,6 +314,15 @@ memory_cache_insert(
     }
 }
 
+void memory_cache_remove(
```
Comment (bdpayne) [inline libvmi/driver/memory_cache.c]: Why is this defined a second time here?

Comment (awsaba) [inline libvmi/driver/memory_cache.c]: I think there has to be 2 versions of these functions, one for with page cache and one without, it's just hard to see the #else at L285.  Maybe a block comment or something around there will help call it out?

Comment (bdpayne) [inline libvmi/driver/memory_cache.c]: Ah yes, you're right.  Funny, I was looking for that else and just scanned right over it.  Must be early still ;-)

**Comment by bdpayne** [comment]
+1, would like to see one more set of eyes before we merge

File: libvmi/driver/memory_cache.c (hunk 1 above)
Comment (tklengyel) [inline libvmi/driver/memory_cache.c]: @awsaba is correct.

**Pull request merged**